src/main/java/com/ratifire/devrate/repository/UserRepository.java

```diff
@@ -22,4 +23,17 @@ public interface UserRepository extends JpaRepository<User, Long> {
   String findPictureByUserId(@Param("id") long id);
 
   List<User> findAllByEventsContaining(Event event);
+
+  @Query("SELECT new com.ratifire.devrate.dto.UserNameSearchDto(u.id, u.firstName, u.lastName) "
+      + "FROM User u "
+      + "WHERE "
+      + "(LOWER(u.firstName) LIKE LOWER(CONCAT('%', :firstName, '%')) "
+      + "AND LOWER(u.lastName) LIKE LOWER(CONCAT('%', :lastName, '%'))) "
+      + "OR (LOWER(u.lastName) LIKE LOWER(CONCAT('%', :firstName, '%')) "
+      + "AND LOWER(u.firstName) LIKE LOWER(CONCAT('%', :lastName, '%'))) "
+      + "OR LOWER(u.firstName) LIKE LOWER(CONCAT('%', :query, '%')) "
+      + "OR LOWER(u.lastName) LIKE LOWER(CONCAT('%', :query, '%'))")
+  List<UserNameSearchDto> findUsersByName(@Param("firstName") String firstName,
```

Comment:
In my opinion, in this case, it makes sense to use Specification to avoid custom query. It will be easier to extend the functionality with adding new filters. What do you think about?